Library/ViewModels/ManagePledgeViewModel.swift

```diff
@@ -64,13 +65,25 @@ public final class ManagePledgeViewModel:
       self.beginRefreshSignal
     )
 
-    let fetchProjectEvent = projectOrParam
+    // Keep track of whether the project has successfully loaded at least once.
+    let projectLoaded = MutableProperty<Bool>(false)
+
+    let shouldFetchProjectWithProjectOrParam = Signal.merge(
+      projectOrParam,
+      projectOrParam.takeWhen(shouldBeginRefresh)
+    )
+
+    let fetchProjectEvent = shouldFetchProjectWithProjectOrParam
+      // Only fetch the project if it hasn't yet succeeded, to avoid this call occurring with each refresh.
+      .filter { [projectLoaded] _ in projectLoaded.value == false }
```

Comment:
🤔 I thought the `.switchMap` took care of the scenario where a request is triggered while another is in-flight?

Reply:
We chatted IRL but just to confirm - this value tracks that the project has been successfully fetched so that if the backing fails to fetch and we pull to refresh, it doesn't try to fetch the project again unnecessarily.